Examples/Run/Reconstruction/Common/RecCKFTracks.cpp

```diff
@@ -77,67 +83,104 @@ int runRecCKFTracks(int argc, char* argv[],
-      std::make_shared<CsvSimHitReader>(simHitReaderCfg, logLevel));
-
-  // Create smeared measurements
-  HitSmearing::Config hitSmearingCfg;
-  hitSmearingCfg.inputSimHits = simHitReaderCfg.outputSimHits;
-  hitSmearingCfg.outputMeasurements = "measurements";
-  hitSmearingCfg.outputSourceLinks = "sourcelinks";
-  hitSmearingCfg.outputMeasurementParticlesMap = "measurement_particles_map";
-  hitSmearingCfg.outputMeasurementSimHitsMap = "measurement_simhits_map";
-  hitSmearingCfg.sigmaLoc0 = 25_um;
-  hitSmearingCfg.sigmaLoc1 = 100_um;
-  hitSmearingCfg.randomNumbers = rnd;
-  hitSmearingCfg.trackingGeometry = trackingGeometry;
-  sequencer.addAlgorithm(
-      std::make_shared<HitSmearing>(hitSmearingCfg, logLevel));
+  // Read the sim hits
+  auto simHitReaderCfg = setupSimHitReading(vm, sequencer);
+  // Read the particles
+  auto particleReader = setupParticleReading(vm, sequencer);
+
+  // Run the sim hits smearing
+  auto hitSmearingCfg = runSimHitSmearing(vm, sequencer, rnd, trackingGeometry,
+                                          simHitReaderCfg.outputSimHits);
 
-  // Pre-select particles
+  // Run the particle selection
   // The pre-selection will select truth particles satisfying provided criteria
   // from all particles read in by particle reader for further processing. It
   // has no impact on the truth hits read-in by the cluster reader.
-  // @TODO: add options for truth particle selection criteria
   TruthSeedSelector::Config particleSelectorCfg;
   particleSelectorCfg.inputParticles = particleReader.outputParticles;
   particleSelectorCfg.inputMeasurementParticlesMap =
       hitSmearingCfg.outputMeasurementParticlesMap;
   particleSelectorCfg.outputParticles = "particles_selected";
-  particleSelectorCfg.ptMin = 1_GeV;
+  particleSelectorCfg.ptMin = 500_MeV;
   particleSelectorCfg.nHitsMin = 9;
   sequencer.addAlgorithm(
       std::make_shared<TruthSeedSelector>(particleSelectorCfg, logLevel));
 
+  // The selected particles
   const auto& inputParticles = particleSelectorCfg.outputParticles;
-  // Create smeared particles states
-  ParticleSmearing::Config particleSmearingCfg;
-  particleSmearingCfg.inputParticles = inputParticles;
-  particleSmearingCfg.outputTrackParameters = "smearedparameters";
-  particleSmearingCfg.randomNumbers = rnd;
-  // Gaussian sigmas to smear particle parameters
-  particleSmearingCfg.sigmaD0 = 20_um;
-  particleSmearingCfg.sigmaD0PtA = 30_um;
-  particleSmearingCfg.sigmaD0PtB = 0.3 / 1_GeV;
-  particleSmearingCfg.sigmaZ0 = 20_um;
-  particleSmearingCfg.sigmaZ0PtA = 30_um;
-  particleSmearingCfg.sigmaZ0PtB = 0.3 / 1_GeV;
-  particleSmearingCfg.sigmaPhi = 1_degree;
-  particleSmearingCfg.sigmaTheta = 1_degree;
-  particleSmearingCfg.sigmaPRel = 0.01;
-  particleSmearingCfg.sigmaT0 = 1_ns;
-  sequencer.addAlgorithm(
-      std::make_shared<ParticleSmearing>(particleSmearingCfg, logLevel));
+
+  // Run particle smearing or run seed finding
+  auto trackParams = [&](bool useTruthSeeds) {
```

Comment:
Does this have to be a lambda? I think this can be just a regular if/else that sets the appropriate variables in the config. Using a lambda here is confusing later on since a simple function call both returns a value and modifies another variables (which one is not clear at the call site). This hides control flow and dependencies.

Reply:
No, it doesn't have to be. All of this have been replaced by the regular if/else.